vis_helpers/pca.py

```diff
@@ -3,104 +3,156 @@
 import streamlit as st
+import peakutils
 
 import processing
 from constants import LABELS
 from . import sidebar, vis_utils
 from visualisation.draw import fig_layout
+from processing import utils
+from vis_helpers.vis_utils import print_widgets_separator
+
 
 def main():
     spectra_types = ['EMPTY', 'BWTEK', 'RENI', 'WITEC', 'WASATCH', 'TELEDYNE', 'JOBIN']
-    spectrometer = st.sidebar.selectbox("Choose spectra type",
-                                        spectra_types,
-                                        format_func=LABELS.get,
-                                        index=0
-                                        )
-
+    
     st.write('## PCA Analysis')
 
-    main_expander = st.beta_expander("Customize your chart")
+    print_widgets_separator(1, sidebar=False)
+    
+    rescale = st.sidebar.checkbox("Rescale")
+    baseline_correction = st.sidebar.checkbox("Baseline correction")
+    
     # Choose plot colors and templates
-    with main_expander:
-        plot_palette, plot_template = vis_utils.get_chart_vis_properties()
-
+    # with main_expander:
+    #     plot_palette, plot_template = vis_utils.get_chart_vis_properties()
+    
     cols = st.beta_columns(2)
-
+    
     with cols[0]:
         groups_num = st.slider('Number of groups', 1, 6, value=2)
         groups_num = int(groups_num)
-        rescale = st.sidebar.checkbox("Rescale")
+    
     with cols[1]:
         components_num = st.slider('Principal components', 1, 3, value=2)
         components_num = int(components_num)
-
+    
+    print_widgets_separator(1, sidebar=False)
+    
     cols = st.beta_columns(groups_num)
     dfs = []
+    group_names = []
     for col_ind, col in enumerate(cols):
         with col:
             files = st.file_uploader(label='Upload data',
                                      accept_multiple_files=True,
                                      type=['txt', 'csv'],
                                      key=f'uploader_{col_ind}'
                                      )
+            
+            spectrometer = st.selectbox("Choose spectra type",
+                                        spectra_types,
+                                        format_func=LABELS.get,
+                                        index=1,
+                                        key=col
+                                        )
+            group_name = st.text_input("Provide group name", f"{col_ind}")
+            group_names.append(group_name)
+            
             if files:
                 df = processing.save_read.files_to_df(files, spectrometer)
                 dfs.append(df)
-
+    
+    print_widgets_separator(1, sidebar=False)
+    
     if len(dfs) < groups_num:
         return
-
+    
     df = pd.concat(dfs, axis=1)
     df = df.interpolate().bfill().ffill()
-    group = [str(i) for i in range(groups_num) for _ in dfs[i].columns]
-
+    group = [group_names[i] for i in range(groups_num) for _ in dfs[i].columns]
```

Comment:
opisz to w komentarzu bo nie czuję co to jest

Reply:
![image](https://user-images.githubusercontent.com/31779294/141981707-acbd62e9-18f0-4b35-9d2c-9f863c010c4a.png)

Reply:
o, smiesznie, mozna przechwycony obraz za pomoca windows+shift+s wkleic i on umie go ogarnac

Reply:
done